File: _src/06/Chapter6.CSharp/Context.cs
using System.Data.Entity;
using Chapter6.CSharp.Maps;
using Chapter6.CSharp.Models;

namespace Chapter6.CSharp
{
    public class Context : DbContext
    {
        public Context()
            : base("name=chapter6")
        {

        }
        public DbSet<Person> People { get; set; }
        public DbSet<Phone> Phones { get; set; }
        public DbSet<PersonType> PersonTypes { get; set; }
        public DbSet<Company> Companies { get; set; }
        public DbSet<PersonViewInfo> PersonView { get; set; }
        protected override void OnModelCreating(DbModelBuilder modelBuilder)
        {
            modelBuilder.Configurations.Add(new PersonMap());
            modelBuilder.Configurations.Add(new PersonTypeMap());
            modelBuilder.Configurations.Add(new PersonViewInfoMap());
            modelBuilder.Configurations.Add(new CompanyMap());
            
        }


    }
}

File: _src/06/Chapter6.VB/Context.vb
Imports System.Data.Entity
Imports System.Data.Entity.ModelConfiguration.Conventions
Imports Chapter6.VB.Models
Imports Chapter6.VB.Maps

Public Class Context
    Inherits DbContext
    Public Sub New()
        MyBase.New("name=chapter6")
    End Sub
    Property People As DbSet(Of Person)
    Property Phones() As DbSet(Of Phone)
    Property PersonTypes() As DbSet(Of PersonType)
    Property Companies() As DbSet(Of Company)
    Property PersonView() As DbSet(Of PersonViewInfo)

    Protected Overrides Sub OnModelCreating(ByVal modelBuilder As DbModelBuilder)
        modelBuilder.Configurations.Add(New PersonMap)
        modelBuilder.Configurations.Add(New PersonTypeMap)
        modelBuilder.Configurations.Add(New PersonViewInfoMap)
        modelBuilder.Configurations.Add(new CompanyMap)
    End Sub


End Class
